### ISSUE TITLE
[Grid][Bug] Screen reader is not announcing name for the 'Clear text' control in the search box present under 'filter pop-up' in Windows.

### ISSUE THREAD

**Issue by msftedad** [telerik/kendo-react#2846]
**Title:** Screen reader is not announcing name for the 'Clear text' control in the search box present under 'filter pop-up' in Windows.

**Test Environment:** 
OS: Windows 11 Enterprise Insider Preview Version 24H2(OS Build 26120.2415)
Browser: Microsoft Edge Version 132.0.2957.26, Chrome Version 131.0.6778.139
URL: [React Data Grid Filtering - KendoReact](https://www.telerik.com/kendo-react-ui/components/grid/filtering#filtering-data-grid-through-column-menu-filter) 
Screen readers: NVDA (2024.4.2), JAWS (Version 2025.2412.50) and Narrator.

**User Impact:**
Screen reader users will find it difficult to understand the control and perform actions if name is not defined for the control.

**Observation:** 
This issue is observed with all three screen readers.

**Pre-Requisite:**
Turn on Screen Readers:
Narrator: Win + Shift + Enter
NVDA: Ctrl + Alt + N
JAWS: Ctrl + Alt + J
Verbosity:
Narrator: Default
JAWS: Beginner, Highest
NVDA: Default

**Repro steps:**

1. Open the URL: [React Data Grid Filtering - KendoReact](https://www.telerik.com/kendo-react-ui/components/grid/filtering#filtering-data-grid-through-column-menu-filter) in latest edge browser.
2. Turn on the screen reader.
3. Press tab key to move to the filter control for the 'Product name' column header in 'Filtering data grid through column menu filter' section and activate it using enter key.
4. Press tab key to move to the 'Clear text' control beside the 'edit field'.
5. Verify on pressing tab key, Screen reader is announcing name for the 'Clear text' control in the search box present under 'filter pop-up' in Windows or not.

**Actual result:**
Screen reader is not announcing name for the 'Clear text' control in the search box present under 'filter pop-up' in Windows. It is announcing it as 'Button'.

**Refer Attachment:** Screen reader is not announcing name for the 'Clear text' control in the search box present under 'filter pop-up' in Windows.wmv

https://github.com/user-attachments/assets/422ba6e6-7ed6-4def-a3ee-d977089b9868

<img width="960" alt="Image" src="https://github.com/user-attachments/assets/2de7f9ca-e26f-4efa-a7be-91afea7fa506" />

**Expected result:**
Screen reader should announce name for the 'Clear text' control in the search box present under 'filter pop-up' in Windows. It should announce it as 'Close button'.

**Comment by PaskalevStoyan** [comment]
Hi @msftedad,

The issue is removed and can be obtained with the latest dev version - `v9.4.0-develop.12`
